src/service/action-impl.js

```diff
@@ -304,61 +311,267 @@ export class ActionService {
     }
-    return actionMap;
+
+  } while (tok.type != TokenType.EOF);
+
+  return actionMap;
+}
+
+
+/**
+ * @enum {number}
+ */
+const TokenType = {
+  INVALID: 0,
+  EOF: 1,
+  SEPARATOR: 2,
+  LITERAL: 3,
+}
+
+/** @private @const {string} */
+const WHITESPACE_SET = ' \t\n\r\f\v\u00A0\u2028\u2029';
+
+/** @private @const {string} */
+const SEPARATOR_SET = ';:.()=,|!';
+
+/** @private @const {string} */
+const STRING_SET = '"\'';
+
+/** @private @const {string} */
+const SPECIAL_SET = WHITESPACE_SET + SEPARATOR_SET + STRING_SET;
+
+
+/** @private */
+class ParserTokenizer {
+  /**
+   * @param {string} str
+   */
+  constructor(str) {
+    /** @private @const {string} */
+    this.str_ = str;
+
+    /** @private {number} */
+    this.index_ = -1;
   }
 
   /**
-   * @param {string} s
-   * @return {?ActionInfoDef}
+   * Returns the next token and advances the position.
+   * @return {!{type: TokenType, value: *}}
    */
-  parseAction_(s) {
-    s = s.trim();
-    if (!s) {
-      return null;
+  next() {
+    const tok = this.next_();
+    this.index_ = tok.index;
+    return tok;
+  }
+
+  /**
+   * Returns the next token but keeps the current position.
+   * @return {!{type: TokenType, value: *}}
+   */
+  peek() {
+    return this.next_();
+  }
+
+  /**
+   * @return {!{type: TokenType, value: *, index: number}}
+   */
+  next_() {
+    let newIndex = this.index_ + 1;
+    if (newIndex >= this.str_.length) {
+      return {type: TokenType.EOF, index: this.index_};
     }
 
-    const eventSep = s.indexOf(':');
-    const methodSep = s.indexOf('.', eventSep + 1);
-    const event = (eventSep != -1 ? s.substring(0, eventSep) : '').toLowerCase()
-        .trim() || null;
-    const target = s.substring(eventSep + 1, methodSep != -1 ? methodSep :
-        s.length).trim();
-    const method = (methodSep != -1 ? s.substring(methodSep + 1) : '')
-        .trim() || DEFAULT_METHOD_;
-
-    if (!event || !target) {
-      log.error(TAG_, 'invalid action definition: ' + s);
-      return null;
+    let c = this.str_.charAt(newIndex);
+
+    // Whitespace: standard set.
+    if (WHITESPACE_SET.indexOf(c) != -1) {
+      newIndex++;
+      for (; newIndex < this.str_.length; newIndex++) {
+        if (WHITESPACE_SET.indexOf(this.str_.charAt(newIndex)) == -1) {
+          break;
+        }
+      }
+      if (newIndex >= this.str_.length) {
+        return {type: TokenType.EOF, index: newIndex};
+      }
+      c = this.str_.charAt(newIndex);
     }
-    return {event: event, target: target, method: method, str: s};
+
+    // A numeric. Notice that it steals the `.` from separators.
+    if (isNum(c) ||
+            c == '.' && newIndex + 1 < this.str_.length &&
+            isNum(this.str_[newIndex + 1])) {
+      let hasFraction = c == '.';
+      let end = newIndex + 1;
+      for (; end < this.str_.length; end++) {
+        const c2 = this.str_.charAt(end);
+        if (c2 == '.') {
+          hasFraction = true;
+          continue;
+        }
```

Comment:
The rule I intended here - if `parseFloat` can parse it - it's all good. But, of course, `parseFloat` is very permissive, so if you'd like - we can tighten this up.